validator/client/validator.go

```diff
@@ -985,13 +985,23 @@ func (v *validator) SetProposerSettings(settings *validatorserviceconfig.Propose
 }
 
 // PushProposerSettings calls the prepareBeaconProposer RPC to set the fee recipient and also the register validator API if using a custom builder.
-func (v *validator) PushProposerSettings(ctx context.Context, km keymanager.IKeymanager) error {
+func (v *validator) PushProposerSettings(ctx context.Context, km keymanager.IKeymanager, overrideDeadline ...time.Duration) error {
 	if km == nil {
 		return errors.New("keymanager is nil when calling PrepareBeaconProposer")
 	}
-
-	deadline := v.SlotDeadline(slots.RoundUpToNearestEpoch(slots.CurrentSlot(v.genesisTime)))
-	ctx, cancel := context.WithDeadline(ctx, deadline)
+	var deadline time.Time
+	if len(overrideDeadline) > 0 && overrideDeadline[0] > 0*time.Second {
```

Comment:
why not just provide the deadline into the main loop ? rather than doing it this way where you have it as an optional parameter

Reply:
updated based on slack discussion on this feedback.